### ISSUE TITLE
Locating the correct vstest.console.exe (VS2017)

### ISSUE THREAD

**Issue by Kemyke** [cake-build/cake#2077]
### What You Are Seeing?
there is a closed issue https://github.com/cake-build/cake/issues/1522 for locating the correct VS2017 `mstest.exe`. But that should be applied when locating vstest.console.exe.

### What is Expected?
If i am right the same modification should be applied to find the correct `vstest.console.exe` [here](https://github.com/cake-build/cake/blob/ad97ab26842fb417189ac36f248f99b14539520c/src/Cake.Common/Tools/VSTest/VSTestRunner.cs).

**Comment by RaringCoder** [comment]
Microsoft have a package:

https://www.nuget.org/packages/Microsoft.TestPlatform/

For deploying the console for VSTest. Should be not be using this as the `#tool` directive for it? I'd rather download a pinned, isolated tool that hope for some console runner on the PC somewhere.

**Comment by jnm2** [comment]
Yes!

**Comment by jnm2** [comment]
Until Cake's VSTest converts to using the new, [officially-recommended](https://developercommunity.visualstudio.com/content/problem/183405/there-is-no-installer-package-id-which-signals-the.html#reply-185103) package https://www.nuget.org/packages/Microsoft.TestPlatform/, this is what I'm doing:

```cake
VSTest(asssemblyPath, new VSTestSettings
{
    // https://github.com/cake-build/cake/issues/2077
    #tool Microsoft.TestPlatform
    ToolPath = Context.Tools.Resolve("vstest.console.exe")
});
```

**Comment by jnm2** [comment]
Ah, see @paulomorgado's https://github.com/cake-build/cake/issues/2152!

**Comment by mholo65** [comment]
#2152 is now merged and functionality for resolving `VSTest` through `Microsoft.TestPlatform` will be available in 0.29.0.

**Comment by jnm2** [comment]
Was this issue meant to be closed?

**Comment by devlead** [comment]
Fixed by #2754
